src/goto-cc/armcc_cmdline.cpp

```diff
@@ -265,19 +267,18 @@ static const char *options_with_arg[]=
   nullptr
 };
 
-static bool
-prefix_in_list(const char *option, const char **list, std::string &prefix)
+static optionalt<std::string>
```

Comment:
Like I said above, I'd prefer not touching this like this for now and instead changing how we store command line arguments, then we don't have to invent our own finds at all.

Reply:
Yes, I think that using a standard container would be a great follow-up refactoring. However I would argue that removing the need for an output parameter is already a step in the right direction, which makes the usages of this function more readable.

Reply:
Done.